src/index.rs

```diff
@@ -403,7 +403,13 @@ impl ResourceIndex {
         log::debug!("Updating a single entry in the index");
 
         if !path.as_ref().exists() {
-            return self.forget_id(old_id);
+            if let Some(indexed_path) = self.id2path.get(&old_id) {
+                if indexed_path.as_path() == path { //FIXME
+                    return self.forget_id(old_id);
+                }
```

Comment:
My Rust skills are not so strong, how to compare paths here? The problem is that `path` is generic...

We need this piece for correct handling absent paths.

Reply:
> My Rust skills are not so strong, how to compare paths here? The problem is that `path` is generic...
> 
> We need this piece for correct handling absent paths.

@kirill. using this code.
if indexed_path.as_path() == path.as_ref() { 
                    let path_buf: PathBuf = indexed_path.into();
                    println!("==indexed_path= {}==path={}==",path_buf.to_str().unwrap_or("no_path_buf"), path.as_ref().to_str().unwrap_or("no_path"));
                    return self.forget_id(old_id);
                }

Reply:
@funnyVOLT thank you!